### PR TITLE
BAEL-5995: Avoiding no runnable methods Error in JUnit

### PR DESCRIPTION
**Pull request by palani-a**
Hello Liam Williams,

	Hope you are doing well!

	I have completed the draft and created this PR:
	Draft Link: https://drafts.baeldung.com/wp-admin/post.php?post=208521&action=edit

	Kindly review and let me know your valuable comments.

	Thanks & Regards
	Palani

### CODE REVIEW

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class JUnit4UnitTest {
+
+    private static int expected = 1;
+
+    @BeforeClass
+    public static void initialize() {
+        System.out.println("Intializes test setup");
```
Comment (theangrydev) [inline testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java:17]: We use SLF4J not System.out in our articles (same elsewhere)

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit5UnitTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @BeforeEach
+    public void initTest() {
+        expected = 2;
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        expected = 0;
```
Comment (theangrydev) [inline testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit5UnitTest.java:32]: What is the purpose of this?

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Before
+    public void initTest() {
+        expected = 2;
+    }
+
+    @After
+    public void cleanUp() {
+        expected = 0;
```
Comment (theangrydev) [inline testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java:32]: What is the purpose of this?

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java (hunk 1 above)
Comment (palani-a) [reply testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java]: Removed this class as it is redundant. Previously, I added this class to refer only to the import statements.

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit5UnitTest.java (hunk 2 above)
Comment (palani-a) [reply testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit5UnitTest.java]: Removed this class as it is redundant. Previously, I added this class to refer only to the import statements.

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java (hunk 3 above)
Comment (palani-a) [reply testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/JUnit4UnitTest.java]: Removed this class as it is redundant. Previously, I added this class to refer only to the import statements.

**Comment by palani-a** [comment]
Hello Liam Williams,

I have fixed your PR comments and re-aligned them with the article. 

Kindly review and let me know your valuable comments.

Thanks & Regards
Palani

**Review by theangrydev — APPROVED** [review]

File: testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/NameUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.baeldung.junit.norunnablemethods;
+
+/**
+ * This class is actually an utility/helper class, 
+ * but incorrectly named so that it could be confused with a real Test class
+ * 
+ * @see @NameutilTestHelper
```
Comment (theangrydev) [inline testing-modules/junit-5-basics-2/src/test/java/com/baeldung/junit/norunnablemethods/NameUtilTest.java:7]:
```suggestion
 * This class is actually a utility/helper class, but it is incorrectly named
 * ending in Test, that can be misinterpreted as a real Test class
 * 
 * @see @NameUtilTestHelper with a name that avoids this problem
```

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Pull request merged**